### PR TITLE
84 merge the 2 chat compts

### PR DESCRIPTION
**Pull request by T1To48**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -40,7 +40,6 @@ const router = createBrowserRouter([
     element: <Chat />,
     errorElement: <>Error...</>,
   },
-  { path: "/profile", element: <ProfilePage /> },
```
Comment (yishain11) [inline client/src/App.jsx:43]: why removed?

Comment (yishain11) [inline client/src/App.jsx:43]: why remove?

File: client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+import { useSelector } from "react-redux";
 import { ChatsContainer } from "../../../styles/Chat/ChatDisplay/ChatsContainer.jsx";
 
-export default function ChatDisplayArea() {
+const ChatDisplayArea = ({ messages }) => {
+  //! must be refactored, when a loggedUser slice is created
+  const { name: loggedUser } = useSelector(
+    (state) => state.userRegister.connectedUser
+  );
+  console.log("loggedUser", loggedUser, "loggedUser");
```
Comment (yishain11) [inline client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx:12]: remove clog

File: client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+
   return (
     <ChatsContainer>
-      {chats.map((message, index) => (
-        // could be sorted by timestamp before display -not added here-
-        <MessageBox message={message} key={index} />
+      {messages.map((message) => (
+        <MessageBox message={message} loggedUser={loggedUser} key={uuidv4()} />
```
Comment (yishain11) [inline client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx:17]: bad practice using rand generated id

File: client/src/main.jsx [hunk 4]
```diff
@@ -8,12 +8,12 @@ import { Provider } from "react-redux";
 import GlobalStyles from "./styles/GlobalStyles.jsx";
 
 ReactDOM.createRoot(document.getElementById("root")).render(
-  <React.StrictMode>
```
Comment (yishain11) [inline client/src/main.jsx:11]: dont remove, this is important for debugging

File: client/src/pages/chat/Chat.page.jsx [hunk 5]
```diff
@@ -16,10 +25,24 @@ const Chat = () => {
...
+
+  useEffect(() => {
+    if (isError) {
+      console.log(error);
+    }
+    if (isSuccess) {
+      console.log(data);
+      setMessages(data.messagesHistory);
+    }
+    console.log(params);
```
Comment (yishain11) [inline client/src/pages/chat/Chat.page.jsx:44]: remove clog

File: client/src/pages/chat/Chat.page.jsx [hunk 6]
```diff
@@ -16,10 +25,24 @@ const Chat = () => {
...
 
+  const { data, isLoading, isSuccess, isError, error } =
+    useGetChatByNamesQuery(usersArr);
+
+  useEffect(() => {
+    if (isError) {
+      console.log(error);
+    }
+    if (isSuccess) {
+      console.log(data);
```
Comment (yishain11) [inline client/src/pages/chat/Chat.page.jsx:41]: remove clog

File: client/src/pages/chat/Chat.page.jsx [hunk 7]
```diff
@@ -16,10 +25,24 @@ const Chat = () => {
...
+  const namesArr = ["tito blah", "bibo mimo"];
+  //!MUST be refactored and replaced when rtk query and chatschema is configured
   //!
 
+  const { data, isLoading, isSuccess, isError, error } =
+    useGetChatByNamesQuery(usersArr);
+
+  useEffect(() => {
+    if (isError) {
+      console.log(error);
```
Comment (yishain11) [inline client/src/pages/chat/Chat.page.jsx:38]: console.error

File: client/src/pages/chat/Chat.page.jsx [hunk 8]
```diff
@@ -40,17 +63,18 @@ const Chat = () => {
...
-        <div key={message.createdAt}>
-          <h2>{message.sender}</h2>
-          <h4>{message.content}</h4>
-        </div>
-      ))}
-    </div>
+    <ChatLayout>
+      <Header reciever={{ name: reciever }} />
+      <ChatDisplayArea messages={messages} />
+      {/* {messages.map((message) => (
```
Comment (yishain11) [inline client/src/pages/chat/Chat.page.jsx:69]: remove commented out code

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx (hunk 3 above)
Comment (T1To48) [reply client/src/components/Chat/ChatDisplayArea/ChatDisplayArea.jsx:17]: but i used the uuid based on the article you sent in slack 
https://react.dev/learn/rendering-lists#where-to-get-your-key

"Locally generated data: If your data is generated and persisted locally (e.g. notes in a note-taking app), use an incrementing counter, [crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) or a package like [uuid](https://www.npmjs.com/package/uuid) when creating items."

**Pull request merged**